docs/reference/mapping/dynamic/templates.asciidoc

```diff
@@ -14,6 +14,10 @@ name
 * <<path-match-unmatch,`path_match` and `path_unmatch`>> operate on the full
 dotted path to the field
 
+* if a dynamic template does not define `match_mapping_type` nor `match` nor
+`path_match`, then it won't match any field except fields are used with 
+name hints in a <bulk, bulk request>.
```

Comment:
I guess it is still possible to have a template with e.g. `match: "geo*"` and point to it by name from a hint?

Reply:
That's correct. I added more tests for this in https://github.com/elastic/elasticsearch/pull/69948/commits/315c8da260c5f9d8f464ad2f31674782af32844a.